### ISSUE TITLE
"Unable to find a suitable cache directory to write to..."

### ISSUE THREAD

**Issue by danyork** [paulej/rfctools#1]
Paul,

Thank you for packaging mmark and xml2rfc in a Docker container. I've followed your instructions and I _seem_ to have everything installed correctly.  However, when I try to use "md2rfc" I get the following error.  I have tried this both with your sample file and also with another file I created.

I _do_ get an XML file... so the mmark side of things seems to work, but then that XML file doesn't seem to be parsed correctly by xml2rfc. 

----

$ docker run --rm --user=$(id -u):$(id -g) -v $(pwd):/rfc -w /rfc paulej/rfctools md2rfc draft-jones-markdown-example-00.md
WARNING: Unable to find a suitible cache directory to write to, trying the following directories:
  /var/cache/xml2rfc
  /.cache/xml2rfc 
Try giving a specific directory with --cache.
WARNING: Unable to find a suitible cache directory to write to, trying the following directories:
  /var/cache/xml2rfc
  /.cache/xml2rfc 
Try giving a specific directory with --cache.
WARNING: Unable to find a suitible cache directory to write to, trying the following directories:
  /var/cache/xml2rfc
  /.cache/xml2rfc 
Try giving a specific directory with --cache.
WARNING: Unable to load the include file at http://xml2rfc.ietf.org/public/rfc/bibxml/reference.RFC.2119.xml
ERROR: Unable to validate the XML document: draft-jones-markdown-example-00.xml
 draft-jones-markdown-example-00.xml: Line 94: Element references content does not follow the DTD, expecting (reference)+, got ()
 draft-jones-markdown-example-00.xml: Line 65: IDREF attribute target references an unknown ID "RFC2119"
Parsing file draft-jones-markdown-example-00.xml

----

Any thoughts on what might be happening here?  I did go and create a /var/cache/xml2rfc directory and make it world-writable.  But that seemed to have no effect.

On a side note, I have had other versions of xml2rfc installed on my system before. Could one of those be conflicting with what exists in the docker container?

Thanks,
Dan

**Comment by danyork** [comment]
Perhaps related to this, when I take the XML file created by "md2rfc" and run it through a version of xml2rfc that I have on my system, I get the error:

----

http://xml2rfc.ietf.org/public/rfc/bibxml/reference.RFC.2119.xml: HTTP/1.1 302 Found

----

**Comment by danyork** [comment]
Hmm... okay, I "fixed" it... but I'm not entirely sure why.

First, I _created_ the file "`$HOME/.cache-xml2rfc`".

And then I ran the command as:

docker run --rm --user=$(id -u):$(id -g) -v $(pwd):/rfc -v $HOME/.cache/xml2rfc:/var/cache/xml2rfc -w /rfc paulej/rfctools md2rfc  <filename>

That worked fine and generated both the XML and text version.

However, if you look at the name of the directory I created it was "`.cache-xml2rfc`" instead of "`.cache/xml2rfc`"... but it worked, even though I had mis-read the instructions.

And if I look in `$HOME` I now have:

- `$HOME/.cache/`
- `$HOME/.cache/xml2rfc`
- `$HOME/.cache-xml2rfc`

So I'm not sure why this worked.

This comment can be closed for me. I'm all set now.  But I'll leave it open in case there is anything here worth noting in the documentation.

**Comment by paulej** [comment]
Dan, I really appreciate you pointing this out to me.  The documentation was lacking in a number of ways (errors as you noted, it did not properly deal with xml2rfc cache requirements, etc.)  I have updated the documentation substantially.  It's not more complex (except that I've added the mapping of the cache directory in each of the commands where it's needed).  Have a look at the new README and see if that clears things up.
